test/catch.cc

```diff
@@ -0,0 +1,232 @@
+#include "eventuals/catch.h"
+
+#include <iostream>
+#include <string>
+
+#include "eventuals/conditional.h"
+#include "eventuals/eventual.h"
+#include "eventuals/expected.h"
+#include "eventuals/just.h"
+#include "eventuals/raise.h"
+#include "eventuals/terminal.h"
+#include "eventuals/then.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using eventuals::Catch;
+using eventuals::Conditional;
+using eventuals::Eventual;
+using eventuals::Expected;
+using eventuals::Just;
+using eventuals::Raise;
+using eventuals::Then;
+using eventuals::Unexpected;
+using testing::MockFunction;
+
+TEST(CatchTest, RaisedRuntimeError) {
+  auto e = []() {
+    return Just(1)
+        | Raise(std::runtime_error("message"))
+        | Catch()
+              .raised<int>([](auto&& error) {
```

Comment:
Instead of using `int` or `double` or `std::string` or any other types for `raised<...>()` let's instead use things like `std::overflow_error` and `std::underflow_error` and anything that extends from `std::exception` as found [here](https://en.cppreference.com/w/cpp/error/exception).

To simplify let's change all the `ADD_FAILURE()` strings to just say `"Encountered unexpected matched raised"`.

Here and the rest of the file please!